### PR TITLE
[RISCV][GISel] Use libcalls for f32/f64 G_FCMP without F/D extensions.

### PR DESCRIPTION
**Pull request by topperc**
LegalizerHelp only supported f128 libcalls and incorrectly assumed that the destination register for the G_FCMP was s32.

### CODE REVIEW

File: llvm/test/CodeGen/RISCV/GlobalISel/double-fcmp.ll [hunk 1]
```diff
@@ -0,0 +1,584 @@
...
+; RV32I-NEXT:    lw ra, 12(sp) # 4-byte Folded Reload
+; RV32I-NEXT:    addi sp, sp, 16
+; RV32I-NEXT:    ret
+;
+; RV64I-LABEL: fcmp_oeq:
+; RV64I:       # %bb.0:
+; RV64I-NEXT:    addi sp, sp, -16
+; RV64I-NEXT:    sd ra, 8(sp) # 8-byte Folded Spill
+; RV64I-NEXT:    call __eqdf2
+; RV64I-NEXT:    slli a0, a0, 32
```
Comment (topperc) [inline llvm/test/CodeGen/RISCV/GlobalISel/double-fcmp.ll:52]: Still need to figure out how to get rid of this shift pair.

File: llvm/test/CodeGen/RISCV/GlobalISel/double-fcmp.ll [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+; RV32I-NEXT:    lw ra, 12(sp) # 4-byte Folded Reload
+; RV32I-NEXT:    addi sp, sp, 16
+; RV32I-NEXT:    ret
+;
+; RV64I-LABEL: fcmp_ogt:
+; RV64I:       # %bb.0:
+; RV64I-NEXT:    addi sp, sp, -16
+; RV64I-NEXT:    sd ra, 8(sp) # 8-byte Folded Spill
+; RV64I-NEXT:    call __gtdf2
+; RV64I-NEXT:    sext.w a0, a0
```
Comment (topperc) [inline llvm/test/CodeGen/RISCV/GlobalISel/double-fcmp.ll:84]: Need to figure out how to get rid of this sext.w

**Review by arsenm — APPROVED** [review]

File: llvm/test/CodeGen/RISCV/GlobalISel/double-fcmp.ll [hunk 3]
```diff
@@ -0,0 +1,599 @@
...
+; RV32I-LABEL: fcmp_false:
+; RV32I:       # %bb.0:
+; RV32I-NEXT:    li a0, 0
+; RV32I-NEXT:    ret
+;
+; RV64I-LABEL: fcmp_false:
+; RV64I:       # %bb.0:
+; RV64I-NEXT:    li a0, 0
+; RV64I-NEXT:    ret
+  %1 = fcmp false double %a, %b
```
[the remark this answers is not in the document]
Comment (topperc) [reply llvm/test/CodeGen/RISCV/GlobalISel/double-fcmp.ll:26]: This was copied from an existing SelectionDAG test. I'd prefer to keep them in sync for easier comparison.

[the remark this answers is not in the document]
Comment (arsenm) [reply llvm/test/CodeGen/RISCV/GlobalISel/double-fcmp.ll:26]: Should just merge the tests with multiple run lines then?

[the remark this answers is not in the document]
Comment (topperc) [reply llvm/test/CodeGen/RISCV/GlobalISel/double-fcmp.ll:26]: Eventually I probably will. This is the first FP IR test I was able to completely copy and get it to pass. I've copied other tests, but had to delete an individual test case for one reason or another that I'll need to come back to.

**Pull request merged**